client/src/app/search/_components/search-menu.tsx

```diff
@@ -0,0 +1,58 @@
+'use client'
+
+import { Icon } from '@iconify/react'
+import Link from 'next/link'
+import { usePathname, useSearchParams } from 'next/navigation'
+
+import { Button } from '@/components/button'
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/dropdown-menu'
+import { Text } from '@/components/text'
+
+const SORT_OPTIONS_MAP: Record<string, string> = {
+  latest: 'ใหม่ล่าสุด',
+  oldest: 'เก่าที่สุด',
+  highest_price: 'ราคาสูงสุด',
+  lowest_price: 'ราคาต่ำสุด',
+}
+
+export function SearchMenu() {
+  const searchParams = useSearchParams()
+  const pathname = usePathname()
+
+  const currentSort = searchParams.get('sort') || 'latest'
+
+  const getSortUrl = (value: string) => {
+    const params = new URLSearchParams(searchParams.toString())
+    params.set('sort', value)
+    return `${pathname}?${params.toString()}`
+  }
+
+  return (
+    <DropdownMenu>
+      <DropdownMenuTrigger asChild>
+        <Button>
+          <Text variant='sm-semibold' desktopVariant='md-semibold'>
+            {SORT_OPTIONS_MAP[currentSort] || 'ใหม่ล่าสุด'}
```

Comment:
ไม่น่ามีเคสที่จะตกไปถึงปะ

```suggestion
            {SORT_OPTIONS_MAP[currentSort]}
```

Reply:
เพื่อ user เข้า /search?sort=xxx

Reply:
ลองแล้วไม่ error อะ แล้วก็คิดว่าถ้าไม่มีก็ไม่ต้อง error ก็ได้รู้สึกว่าไม่ได้ critical ขนาดนั้น